StatsdClient/ClientConfig.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Configuration;
+
+namespace StatsdClient
+{
+  /// <summary>
+  /// Configuration Element for reading statsdClient settings from a .config file
+  /// </summary>
+  public class ClientConfigElement : ConfigurationElement
+  {
+    /// <summary>
+    /// The hostname or IP address of the statsd server
+    /// </summary>
+    [ConfigurationProperty("host", IsRequired = true)]
+    public string Host
+    {
+      get { return this["host"] as string; }
+      set { this["host"] = value; }
+    }
+
+    /// <summary>
+    /// The port statsd is listening on
+    /// </summary>
+    [ConfigurationProperty("port", IsRequired = true)]
+    public int Port
+    {
+      get { return (int)this["port"]; }
+      set { this["port"] = value; }
+    }
+
+    /// <summary>
+    /// The prefix to prepend to all metrics
+    /// </summary>
+    [ConfigurationProperty("prefix", IsRequired = false, DefaultValue = "Test")]
+    public string Prefix
+    {
+      get { return this["prefix"] as string; }
+      set { this["prefix"] = value; }
+    }
+
+    /// <summary>
+    /// The connection type (Udp/Tcp)
+    /// </summary>
+    [ConfigurationProperty("connType", IsRequired = false, DefaultValue = ConnectionType.Udp)]
+    public ConnectionType ConnectionType
+    {
+      get { return (ConnectionType)Enum.Parse(typeof(ConnectionType), this["connType"].ToString()); }
+      set { this["connType"] = value; }
+    }
+
+    /// <summary>
+    /// Retry the connection if it fails (TCP only).
+    /// </summary>
+    [ConfigurationProperty("tcpRetryOnDisconnect", IsRequired = false, DefaultValue = true)]
+    public bool RetryOnDisconnect
```

Comment:
I don't like this name. This is very specific to tcp implementation. We could possibly use this value to control if we retry other things in the lib like udp (socket exceptions..) I'm not sure of a good name, perhaps CanRetry?? o